pkg/phlaredb/query/iters.go

```diff
@@ -96,17 +95,188 @@ func (t RowNumber) Valid() bool {
 // gb     | 1 | 3 | {  0,  2,  0,  0 }
 // null   | 0 | 1 | {  1,  0, -1, -1 }
 func (t *RowNumber) Next(repetitionLevel, definitionLevel int) {
-	// Next row at this level
 	t[repetitionLevel]++
 
-	// New children up through the definition level
-	for i := repetitionLevel + 1; i <= definitionLevel; i++ {
-		t[i] = 0
-	}
-
-	// Children past the definition level are undefined
-	for i := definitionLevel + 1; i < len(t); i++ {
-		t[i] = -1
+	// the following is nextSlow() unrolled
+	switch repetitionLevel {
```

Comment:
Do you know what's going on here ? why 5 ?

Reply:
I think might be the highest definition level the otel schema has? 

Probably @joe-elliott might be able to anwser that better.

In the grand scheme of things, the gain is probably not worth the unrolling for us, happy to put that back to the original (https://github.com/grafana/tempo/pull/2412).

Reply:
@simonswine is correct. it's just an arbitrary depth that is enough for our schema.

i don't recall the numbers but profiling heavy "needle in the haystack" queries showed a large %age of the time in this function (which is why we unrolled it).

Reply:
I'm fine if we don't overflow our current definition/repetition level for the 3 column we use this iterator with.

Reply:
It is inline with limit  imposed by `RowNumber [6]int64§